splitio/api/commons.py

```diff
@@ -113,4 +115,41 @@ def build_fetch(change_number, fetch_options, metadata):
         extra_headers[_CACHE_CONTROL] = _CACHE_CONTROL_NO_CACHE
     if fetch_options.change_number is not None:
         query['till'] = fetch_options.change_number
-    return query, extra_headers
\ No newline at end of file
+    return query, extra_headers
+
+class APIException(Exception):
+    """Exception to raise when an API call fails."""
+
+    def __init__(self, custom_message, status_code=None):
+        """Constructor."""
+        Exception.__init__(self, custom_message)
+        self._status_code = status_code if status_code else -1
+
+    @property
+    def status_code(self):
+        """Return HTTP status code."""
+        return self._status_code
+
+class HttpClientException(Exception):
+    """HTTP Client exception."""
+
+    def __init__(self, message):
+        """
+        Class constructor.
+
+        :param message: Information on why this exception happened.
+        :type message: str
+        """
+        Exception.__init__(self, message)
+
+def build_basic_headers(apikey):
```

Comment:
same goes for this function, it should only be used by the HTTP client, so it's better off scoped to that module

Reply:
will move